packaging/dependencies/amazon-cloudwatch-agent-ctl

```diff
@@ -249,7 +249,7 @@ main() {
     # detect which init system is in use
     if [ "$(/sbin/init --version 2>/dev/null | grep -c upstart)" = 1 ]; then
 	SYSTEMD='false'
-    elif [ "$(systemctl | grep -c '\-\.mount')" = 1 ]; then
+    elif [ "$(systemctl | grep -c -E '\s\-\.mount\s')" = 1 ]; then
```

Comment:
shall we change `\s` to `\s*`? not sure if the number of whitespace will change in the future.

Or other solution is to exclude `tmp-.mount_jetbra`,  the I am not sure what the exactly output systemctl should be, but looks like current fix has some backward compatible issue.

Reply:
> shall we change `\s` to `\s*`? not sure if the number of whitespace will change in the future.
> 

I don't think so, `'\s\-\.mount\s'` will match to any extra symbols at the start or at the end because it doesn't contain line start/end markers "^$".

> Or other solution is to exclude `tmp-.mount_jetbra`, the 

Also, I don't think that we should rely on any 3rd party apps behavior in this script

> I am not sure what the exactly output systemctl should be, but looks like current fix has some backward compatible issue.

We can omit the first space detector in the pattern and use this one `'\-\.mount\s'` - it will mitigate the risk of backward incompatibility.

Reply:
> > shall we change `\s` to `\s*`? not sure if the number of whitespace will change in the future.
> 
> I don't think so, `'\s\-\.mount\s'` will match to any extra symbols at the start or at the end because it doesn't contain line start/end markers "^$".
> 
> > Or other solution is to exclude `tmp-.mount_jetbra`, the
> 
> Also, I don't think that we should rely on any 3rd party apps behavior in this script
> 
> > I am not sure what the exactly output systemctl should be, but looks like current fix has some backward compatible issue.
> 
> We can omit the first space detector in the pattern and use this one `'\-\.mount\s'` - it will mitigate the risk of backward incompatibility.

I tested `'\s\-\.mount\s'` does not work.   you can use '\-\.mount\s'

Reply:
Changed it